internals-js/src/specs/listSizeSpec.ts

```diff
@@ -0,0 +1,40 @@
+import { DirectiveLocation } from 'graphql';
+import { createDirectiveSpecification } from '../directiveAndTypeSpecification';
+import { FeatureDefinition, FeatureDefinitions, FeatureUrl, FeatureVersion } from './coreSpec';
+import { ListType, NonNullType } from '../definitions';
+import { registerKnownFeature } from '../knownCoreFeatures';
+import { ARGUMENT_COMPOSITION_STRATEGIES } from '../argumentCompositionStrategies';
+
+export const listSizeIdentity = 'https://specs.apollo.dev/listSize';
+
+export class ListSizeSpecDefinition extends FeatureDefinition {
+  constructor(version: FeatureVersion, readonly minimumFederationVersion: FeatureVersion) {
+    super(new FeatureUrl(listSizeIdentity, 'listSize', version), minimumFederationVersion);
+
+    this.registerDirective(createDirectiveSpecification({
+      name: 'listSize',
+      locations: [DirectiveLocation.FIELD_DEFINITION],
+      args: [
+        { name: 'assumedSize', type: (schema) => schema.intType(), compositionStrategy: ARGUMENT_COMPOSITION_STRATEGIES.NULLABLE_MAX },
+        { name: 'slicingArguments', type: (schema) => new ListType(new NonNullType(schema.stringType())), compositionStrategy: ARGUMENT_COMPOSITION_STRATEGIES.NULLABLE_UNION },
+        { name: 'sizedFields', type: (schema) => new ListType(new NonNullType(schema.stringType())), compositionStrategy: ARGUMENT_COMPOSITION_STRATEGIES.NULLABLE_UNION },
+        { name: 'requireOneSlicingArgument', type: (schema) => schema.booleanType(), defaultValue: true, compositionStrategy: ARGUMENT_COMPOSITION_STRATEGIES.NULLABLE_AND },
```

Comment:
Since there will always be a default value, should we use just a regular AND?

Reply:
It looks like the types don't quite match up in this case. I need to use `schema.booleanType()` here so the directive definition matches the spec (ie. the directive has `requireOneSlicingArgument: Boolean = true`). The argument merger doesn't do any auto-upgrading if it has a nullable type with a default value, so it has to handle `schema.booleanType()` which implies null/undefined handling. Setting the merger's require type to `new NonNullType(schema.booleanType())` results in a type mismatch error during composition.